### ISSUE TITLE
find: File system loop detected

### ISSUE THREAD

**Issue by mastertinner** [pivotal-cf/bookbinder#119]
When binding my book, I keep getting `find: File system loop detected` and it won't bind... It doesn't provide any helpful error message. Does anybody know what the problem could be? I'm running bookbinder in a Docker container based on the `ruby:2.3` image. My book is https://github.com/swisscom/docs-appcloud-book.

Here's the full stack trace:

```txt
       error  build/deploying/openstack/cf-stub.html
/usr/local/bundle/gems/git-1.2.9.1/lib/git/lib.rb:917:in `command'
/usr/local/bundle/gems/git-1.2.9.1/lib/git/lib.rb:76:in `clone'
/usr/local/bundle/gems/git-1.2.9.1/lib/git/base.rb:29:in `clone'
/usr/local/bundle/gems/git-1.2.9.1/lib/git.rb:97:in `clone'
/usr/local/bundle/gems/bookbindery-9.12.0/lib/bookbinder/ingest/git_accessor.rb:97:in `cached_clone'
/usr/local/bundle/gems/bookbindery-9.12.0/lib/bookbinder/ingest/git_accessor.rb:15:in `clone'
/usr/local/bundle/gems/bookbindery-9.12.0/lib/bookbinder/ingest/git_cloner.rb:18:in `call'
/app/output/master_middleman/bookbinder_helpers.rb:31:in `yield_for_code_snippet'
/app/output/master_middleman/source/deploying/openstack/cf-stub.html.md.erb:24:in `block in singleton class'
/app/output/master_middleman/source/deploying/openstack/cf-stub.html.md.erb:-6:in `instance_eval'
/app/output/master_middleman/source/deploying/openstack/cf-stub.html.md.erb:-6:in `singleton class'
/app/output/master_middleman/source/deploying/openstack/cf-stub.html.md.erb:-8:in `__tilt_7401700'
/usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:170:in `call'
/usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:170:in `evaluate'
/usr/local/bundle/gems/tilt-1.4.1/lib/tilt/template.rb:103:in `render'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/rendering.rb:313:in `render_individual_file'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/rendering.rb:181:in `_render_with_all_renderers'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/rendering.rb:149:in `render_template'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/sitemap/resource.rb:127:in `block in render'
/usr/local/bundle/gems/activesupport-4.2.9/lib/active_support/notifications.rb:166:in `instrument'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/util.rb:41:in `instrument'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/application.rb:244:in `instrument'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/sitemap/resource.rb:14:in `instrument'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/sitemap/resource.rb:100:in `render'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:260:in `process_request'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:210:in `block in call!'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:209:in `catch'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:209:in `call!'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/core_extensions/request.rb:195:in `call'
/usr/local/bundle/gems/rack-1.6.8/lib/rack/urlmap.rb:66:in `block in call'
/usr/local/bundle/gems/rack-1.6.8/lib/rack/urlmap.rb:50:in `each'
/usr/local/bundle/gems/rack-1.6.8/lib/rack/urlmap.rb:50:in `call'
/usr/local/bundle/gems/rack-1.6.8/lib/rack/head.rb:13:in `call'
/usr/local/bundle/gems/rack-1.6.8/lib/rack/lint.rb:49:in `_call'
/usr/local/bundle/gems/rack-1.6.8/lib/rack/lint.rb:37:in `call'
/usr/local/bundle/gems/rack-1.6.8/lib/rack/builder.rb:153:in `call'
/usr/local/bundle/gems/rack-test-0.7.0/lib/rack/mock_session.rb:30:in `request'
/usr/local/bundle/gems/rack-test-0.7.0/lib/rack/test.rb:249:in `process_request'
/usr/local/bundle/gems/rack-test-0.7.0/lib/rack/test.rb:59:in `get'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli/build.rb:257:in `render_to_file'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli/build.rb:223:in `build_resource'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli/build.rb:215:in `each'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli/build.rb:215:in `execute!'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli/build.rb:128:in `invoke!'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/actions.rb:95:in `action'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli/build.rb:70:in `build'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
/usr/local/bundle/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/base.rb:444:in `start'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli.rb:72:in `method_missing'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/command.rb:29:in `run'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/command.rb:126:in `run'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
/usr/local/bundle/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
/usr/local/bundle/gems/thor-0.19.4/lib/thor/base.rb:444:in `start'
/usr/local/bundle/gems/middleman-core-3.4.1/lib/middleman-core/cli.rb:20:in `start'
/usr/local/bundle/gems/middleman-core-3.4.1/bin/middleman:18:in `<top (required)>'
/usr/local/bundle/bin/middleman:17:in `load'
/usr/local/bundle/bin/middleman:17:in `<main>'
find: File system loop detected; `/app/output/master_middleman/source/cf-release/releases/cf' is part of the same file system loop as `/app/output/master_middleman/source/cf-release/releases'.

find: File system loop detected; `/app/output/master_middleman/source/cf-release/src/capi-release/src/capi-release/src' is part of the same file system loop as `/app/output/master_middleman/source/cf-release/src/capi-release/src'.

find: File system loop detected; `/app/output/master_middleman/source/cf-release/releases/cf' is part of the same file system loop as `/app/output/master_middleman/source/cf-release/releases'.

find: File system loop detected; `/app/output/master_middleman/source/cf-release/src/capi-release/src/capi-release/src' is part of the same file system loop as `/app/output/master_middleman/source/cf-release/src/capi-release/src'.

Your bind failed. Rerun with --verbose to troubleshoot.
```

**Comment by pspinrad** [comment]
Hey @mastertinner -- I was getting same / similar behavior recently.  Apparently `cf-release` has a loop in its links or symlinks, and bookbinder got caught in it because I had that repo cloned locally.  When I renamed my local `cf-release` directory to something else (well, actually, when Max renamed it), bookbinder worked.  Hopefully this easy fix will work for you too.

**Comment by mastertinner** [comment]
Thanks, @pspinrad. I don't think that I'm having the same problem as what you described. I was always building in a Docker image with no sibling directories whatsoever... Since I changed one of the variables in my `config/template_variables.yml` it works again... I also switched to using `bind local` instead of `bind remote` which seems to have made it more stable as well. I don't know how or why but I'm not getting the above problem anymore so I'll close this issue for now.
